### PR TITLE
Fixed (Iss #414): `Inferer` now uses `torch.no_grad()` correctly

### LINKED ISSUE #414: [MAINT] Inferer.__call__() function in solaris.nets.infer.py does not properly use torch.no_grad()
## Maintenance request summary

Inferer.__call__() function in solaris.nets.infer.py does not properly use torch.no_grad(). PyTorch is effectively computing gradient for the inference process, thus, using significantly more GPU memory.

## Task detail and notes

Any additional information:

- This code change will not impact anything. It will only improve GPU memory efficiency.

### PR DESCRIPTION
**Pull request by imadtoubal**
# Description

`Inferer.__call__()` function in `solaris/nets/infer.py` does not properly use `torch.no_grad()`. PyTorch is effectively computing gradient for the inference process, thus, using significantly more GPU memory.

Fixes #414 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] __Breaking change__ (fix or feature that would cause existing functionality to not work as expected - these changes will not be merged until major releases!)

# How Has This Been Tested?

Please describe tests that you added to the pytest codebase (if applicable).

# Checklist:

- [x] My PR has a descriptive title
- [x] My code follows PEP8
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas (NA)
- [ ] I have made corresponding changes to the documentation (NA)
- [x] My changes generate no new errors
- [ ] I have added tests that prove my fix is effective or that my feature works (NA)
- [x] My PR passes Travis CI tests
- [x] My PR does not reduce coverage in Codecov

_If your PR does not fulfill all of the requirements in the checklist above, that's OK!_ Just prepend [WIP] to the PR title until they are all satisfied. If you need help, @-mention a maintainer and/or add the __Status: Help Needed__ label.

### CODE REVIEW

**Comment by rbavery** [comment]
closing since this relates to pytorch and this library no longer covers training or inference related functionality https://github.com/CosmiQ/solaris/blob/32236217bd913fbb238e19f2f8420e5794fd8136/ROADMAP.md
